cmd/libs/go2idl/deepcopy-gen/generators/deepcopy.go

```diff
@@ -0,0 +1,388 @@
+					return generators
+				},
+				FilterFunc: func(c *generator.Context, t *types.Type) bool {
+					switch t.Kind {
+					case types.Func, types.Chan:
+						// These types can't be copied.
+						return false
+					case types.Unknown, types.Unsupported:
+						// These types are explicitly ignored.
+						return false
+					case types.Array:
+						// We don't support arrays.
+						return false
+					}
+					// Also, filter out private types.
+					// FIXME: Do we want to filter them?
+					if strings.ToLower(t.Name.Name[:1]) == t.Name.Name[:1] {
+						return false
+					}
+					return true
+				},
+			})
+	}
+	return packages
+}
+
+/*func isCopyable(seen map[*types.Type]bool, t *types.Type) bool {
+	if seen[t] {
+		// be optimistic in the case of type cycles.
+		return true
+	}
+	seen[t] = true
+	if t.Name.Package != "" {
+		// Can't deep copy non-anonymous, private types.
+		if strings.ToLower(t.Name.Name[:1]) == t.Name.Name[:1] {
+			return false
+		}
+	}
+	switch t.Kind {
+	case types.Builtin:
+		return true
+	case types.Slice, types.Pointer:
+		return isCopyable(seen, t.Elem)
+	case types.Map:
+		return isCopyable(seen, t.Key) && isCopyable(seen, t.Elem)
+	case types.Struct:
+		for _, m := range t.Members {
+			if strings.ToLower(m.Name[:1]) == m.Name[:1] {
+				return false
+			}
+			if !isCopyable(seen, m.Type) {
+				return false
+			}
+		}
+		return true
+	case types.Func, types.Chan:
+		return false
+	default:
+		return false
+	}
+}*/
+
+const (
+	apiPackagePath = "k8s.io/kubernetes/pkg/api"
+	conversionPackagePath = "k8s.io/kubernetes/pkg/conversion"
+)
+
+// genDeepCopy produces a file with a set for a single type.
+type genDeepCopy struct {
+	generator.DefaultGen
+	targetPackage string
+	imports       *generator.ImportTracker
+	typesForInit  []*types.Type
+}
+
+func NewGenDeepCopy(sanitizedName, targetPackage string) generator.Generator {
+	return &genDeepCopy{
+		DefaultGen: generator.DefaultGen{
+			OptionalName: sanitizedName,
+		},
+		targetPackage: targetPackage,
+		imports: generator.NewImportTracker(),
+		typesForInit: make([]*types.Type, 0),
+	}
+}
+
+func (g *genDeepCopy) Namers(c *generator.Context) namer.NameSystems {
+	// Have the raw namer for this file track what it imports.
+	return namer.NameSystems{"raw": namer.NewRawNamer(g.imports)}
+}
+
+// Filter ignores all but one type because we're making a single file per type.
+func (g *genDeepCopy) Filter(c *generator.Context, t *types.Type) bool {
+	// Filter out all types not copyable within the package.
+	copyable := g.copyableWithinPackage(t)
+	if copyable {
+		g.typesForInit = append(g.typesForInit, t)
+	}
+	return copyable
+}
+
+func (g *genDeepCopy) copyableWithinPackage(t *types.Type) bool {
+	// TODO: We should also generate functions for other types that
+	// we are referencing. This can be done either by making functions
+	// in other packages public, or by generating everything everywhere.
```

Comment:
In principle, I agree that there is no reason.
However, since we already have an existing generator (not using the framework) - I just wanted to migrate it to the framework with this PR with minimal number of changes.
Once this is done, it should be much easier to change things (e.g. making those functions public and not generate everything everywhere...)
@pmorie Does it sound reasonable to you?